packages/realm_dart/lib/src/native/realm_core.dart

```diff
@@ -1794,16 +1794,47 @@ class _RealmCore {
     return RealmNotificationTokenHandle._(pointer, list.realm.handle);
   }
 
-  RealmNotificationTokenHandle subscribeObjectNotifications(RealmObjectBase object, NotificationsController controller) {
-    final pointer = _realmLib.invokeGetPointer(() => _realmLib.realm_object_add_notification_callback(
-          object.handle._pointer,
-          controller.toPersistentHandle(),
-          _realmLib.addresses.realm_dart_delete_persistent_handle,
-          nullptr,
-          Pointer.fromFunction(object_change_callback),
-        ));
+  RealmNotificationTokenHandle subscribeObjectNotifications(RealmObjectBase object, NotificationsController controller,
+      [List<String>? keyPaths, int? classKey]) {
+    return using((Arena arena) {
+      Pointer<realm_key_path_array> kpNative = nullptr;
+
+      if (keyPaths != null && classKey != null) {
+        final length = keyPaths.length;
+        final keypathsNative = arena<Pointer<Char>>(length);
+
+        for (int i = 0; i < length; i++) {
+          keypathsNative[i] = keyPaths[i].toCharPtr(arena);
+        }
+
+        kpNative = _realmLib.invokeGetPointer(() => _realmLib.realm_create_key_path_array(object.realm.handle._pointer, classKey, length, keypathsNative));
+      }
```

Comment:
LGTM

Perhaps get the `classKey` from the `realmObject` instead of passing it as an argument?